idaes/core/util/tables.py

```diff
@@ -235,10 +235,18 @@ class VariableTypes:
 
     for key, sb in stream_states.items():
         stream_attributes[key] = {}
-        if true_state:
-            disp_dict = sb.define_state_vars()
-        else:
-            disp_dict = sb.define_display_vars()
+        try:
+            if true_state:
+                disp_dict = sb.define_state_vars()
+            else:
+                disp_dict = sb.define_display_vars()
+        except AttributeError:  # if define_(display,state)_vars doesn't exist(!)
```

Comment:
Perhaps refactor to
```
try:
    if true_state:
        disp_func = sb.define_state_vars
    else:
        disp_func = sb.define_display_vars
except AttributeError:
   disp_func = None
   ....
if disp_func is not None:
    disp_dict = disp_func()
else:
    disp_dict = {}
```

That way we catch only the attribute error for the function not existing, but not attribute errors internal to the function.

Reply:
Actually it looks like you can just use `getattr` and provide a default to avoid the `try`-`except` block. I'm not sure whether `None` would work as a default, though.